client/blocks/support-article-dialog/dialog.jsx

```diff
@@ -1,55 +1,60 @@
 import { Button, Dialog, Gridicon } from '@automattic/components';
-import { isDefaultLocale } from '@automattic/i18n-utils';
 import { useTranslate } from 'i18n-calypso';
 import { memoize } from 'lodash';
-import PropTypes from 'prop-types';
 import { useEffect } from 'react';
-import { connect } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import { SUPPORT_BLOG_ID } from 'calypso/blocks/inline-help/constants';
 import QueryReaderPost from 'calypso/components/data/query-reader-post';
 import QueryReaderSite from 'calypso/components/data/query-reader-site';
-import QuerySupportArticleAlternates from 'calypso/components/data/query-support-article-alternates';
 import EmbedContainer from 'calypso/components/embed-container';
+import useSupportArticleAlternatesQuery from 'calypso/data/support-article-alternates/use-support-article-alternates-query';
 import { useRouteModal } from 'calypso/lib/route-modal';
-import { closeSupportArticleDialog as closeDialog } from 'calypso/state/inline-support-article/actions';
+import { closeSupportArticleDialog } from 'calypso/state/inline-support-article/actions';
 import { getPostByKey } from 'calypso/state/reader/posts/selectors';
-import getCurrentLocaleSlug from 'calypso/state/selectors/get-current-locale-slug';
 import getCurrentQueryArguments from 'calypso/state/selectors/get-current-query-arguments';
 import getInlineSupportArticleActionIsExternal from 'calypso/state/selectors/get-inline-support-article-action-is-external';
 import getInlineSupportArticleActionLabel from 'calypso/state/selectors/get-inline-support-article-action-label';
 import getInlineSupportArticleActionUrl from 'calypso/state/selectors/get-inline-support-article-action-url';
 import getInlineSupportArticleBlogId from 'calypso/state/selectors/get-inline-support-article-blog-id';
 import getInlineSupportArticlePostId from 'calypso/state/selectors/get-inline-support-article-post-id';
-import {
-	getSupportArticleAlternatesForLocale,
-	isRequestingSupportArticleAlternates,
-	shouldRequestSupportArticleAlternates,
-} from 'calypso/state/support-articles-alternates/selectors';
 import SupportArticleHeader from './header';
 import Placeholders from './placeholders';
 
 import './style.scss';
 import './content.scss';
 
 const noop = () => {};
+const getPostKey = memoize(
+	( blogId, postId ) => ( { blogId, postId } ),
+	( ...args ) => JSON.stringify( args )
+);
```

Comment:
Seems to me this doesn't need to be memoized any more. The result used to be a `postKey` prop injected by `mapStateToProps`, but nowadays it's just a local variable in the component.

Reply:
That's fair. Memoization removed in 195587c31f3310b0d0eae18de15185609725a58b.